### PR TITLE
feat(photon os): added EOL dates check

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description

- Added [end-of-life dates](https://blogs.vmware.com/vsphere/2022/01/photon-1-x-end-of-support-announcement.html) for Photon os.

### CODE REVIEW

File: pkg/detector/ospkg/photon/photon.go [hunk 1]
```diff
@@ -61,7 +89,12 @@ func (s *Scanner) Detect(osVer string, pkgs []ftypes.Package) ([]types.DetectedV
 	return vulns, nil
 }
 
-// IsSupportedVersion checks is OSFamily can be scanned
-func (s *Scanner) IsSupportedVersion(_, _ string) bool {
-	return true
+// IsSupportedVersion checks is OSFamily can be scanned with Photon scanner
```
Comment (knqyf263) [inline pkg/detector/ospkg/photon/photon.go:92]:
```suggestion
// IsSupportedVersion checks if the OS version reached end-of-support.
```

File: pkg/detector/ospkg/photon/photon.go [hunk 2]
```diff
@@ -11,20 +14,45 @@ import (
...
-// EOL can't be found for photon https://github.com/vmware/photon/issues/1031
-//var (
-//	eolDates = map[string]time.Time{}
-//)
+var (
+	eolDates = map[string]time.Time{
+		"1.0": time.Date(2022, 2, 28, 23, 59, 59, 0, time.UTC),
+		"2.0": time.Date(2022, 12, 31, 23, 59, 59, 0, time.UTC),
+		"3.0": time.Date(2024, 6, 30, 23, 59, 59, 0, time.UTC),
+		"4.0": time.Date(2025, 12, 31, 23, 59, 59, 0, time.UTC),
```
Comment (knqyf263) [inline pkg/detector/ospkg/photon/photon.go:22]: Need comments

**Review by knqyf263 — CHANGES_REQUESTED** [review]

File: pkg/detector/ospkg/photon/photon.go (hunk 2 above)
Comment (DmitriyLewen) [reply pkg/detector/ospkg/photon/photon.go:24]: comment added

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
@DmitriyLewen Would you resolve conflicts?

**Comment by DmitriyLewen** [comment]
@knqyf263 conflict resolved

**Comment by knqyf263** [comment]
Thanks

**Pull request merged**